File: backend/dataset/reverse/reverse_9.py
class Solution:
    def reverse(self, x: int) -> int:
        isneg = x<0
        x = abs(x)
        rev = 0
        while(x>0):
            rem = x%10
            rev = rev*10+rem
            x = x//10
        
        if rev >= 2** 31 -1 or rev <= -2** 31:
            return 0

        elif isneg:
            return 0-rev
        else:
            return rev 

File: backend/CDataset/reverse/reverse_9.cpp
class Solution {
public:
    int reverse(int x) {
        int newX = 0;
        
        while (x != 0) {
            if (newX > (INT_MAX-abs(x%10))/10 || newX < (INT_MIN+abs(x%10))/10) 
                return 0;
            newX = newX*10 + x%10;
            x /= 10;
        }
        
        return newX;
    }   
};
